src/main/java/mate.academy/lib/Injector.java

```
@@ -1,13 +1,78 @@
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        checkAnnotation(clazz);
+        Object implementationInstance = createNewInstance(clazz);
+        Field[] fields = clazz.getDeclaredFields();
+        for (Field field : fields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                try {
+                    field.setAccessible(true);
+                    field.set(implementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Failed to initialize field value in class: "
+                            + clazz.getName()
+                            + " and field: "
+                            + field.getName(), e);
+                }
+            }
+        }
+        return implementationInstance;
+    }
+
+    private void checkAnnotation(Class<?> clazz) {
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Injection failed. "
+                    + "The @Component annotation is missing for the "
+                    + clazz.getName());
+        }
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException("Failed to create a new instance of "
+                    + clazz.getName(), e);
+        }
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        if (interfaceClazz.isInterface() && interfaceClazz != null) {
+            return implementations.get(interfaceClazz);
```

Issue: If the interface is not present in the `implementations` map, `implementations.get(interfaceClazz)` will return null, which can lead to a NullPointerException later in the code. According to the requirements, you should throw a meaningful exception if no implementation is found for the interface. Please update this logic to check for null and throw an appropriate exception if the implementation is missing.